### PR TITLE
Add start of page describing types of data expected in Simperium

### PR DESCRIPTION
**Pull request by dmsnell**
This is just the beginning, but here are some definitions constraining the data types that we can send or expect to receive from Simperium the server.

### CODE REVIEW

File: TYPES.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+There are certain types of data that Simperium uses which have specific constraints. Check here for those data types and constraints.
+
+## Basic types
+
+| type | min. length | max length | allowable values | notes |
+|---|:-:|:-:|---|---|
+| **app id** | | | `a-z` `A-Z` `0-9` `.` `_` `-` | case insensitive |
+| **bucket name** | 1 | 64 | `a-z` `A-Z` `0-9` `.` `_` `-` `%` | |
+| **ccid** | 1 | 256 | `a-f` `A-F` `0-9` | [type 4 (random) UUID](https://tools.ietf.org/html/rfc4122) in hex |
+| **change type** | | | `M` `-` `DROP` `EMPTY` | see _change types_ below|
```
Comment (beaucollins) [inline TYPES.md]: Where do you get `DROP` and `EMPTY`? I haven't seen those before.

Comment (dmsnell) [reply TYPES.md]: They come from the Python source for the server. Nice catch!

Comment (beaucollins) [reply TYPES.md]: I see. Those come from the array json diff operations that are still not being used I think.

For change operation objects I believe the only valid change types are: `+`, `-` and `M` (add object to bucket, remove object from bucket, and modify existing object in bucket).

Comment (beaucollins) [reply TYPES.md]: It's a subset of what jsondiff supports.

Comment (dmsnell) [reply TYPES.md]: as change operations:
 - `DROP` means that the bucket should be removed. this is an admin-only operation
 - `EMPTY` means that the bucket should be purged of all its keys

**Review by beaucollins — APPROVED** [review]

File: TYPES.md [hunk 2]
```diff
@@ -9,7 +9,45 @@ There are certain types of data that Simperium uses which have specific constrai
...
 | **client id** | 1 | 256 | | |
 | **entity key** | 1 | 256 | `a-z` `A-Z` `0-9` `.` `_` `-` `%` `@` | |
 | **entity version** | | | integer | |
+
+## Change types
+
+There are four primary change types available to the Simperium protocol.
+Of these, `DROP` is special in that it requires administrative privileges on the bucket to be able to successfully execute.
+
+### Modify entity - `M`
```
Comment (beaucollins) [inline TYPES.md:22]: The preferred term was changed from `entity` to `object`. Simperium pre-websockets used the `entity` vocab so there are remnants of it in source.

**Comment by dmsnell** [comment]
@beaucollins hope you don't mind but I'm going ahead and merging this to clear it from my queue. it's obviously incomplete but I feel like it's not inaccurate as-is. Please let me know if you disagree or want me to clean up anything here

**Pull request merged**